CompuCell3D/core/CompuCell3D/Boundary/BoundaryStrategy.cpp

```diff
@@ -84,9 +86,17 @@ BoundaryStrategy::BoundaryStrategy(string boundary_x, string boundary_y,
 
 {
 
+    boundaryConditionIndicator.assign(3, 0);
+
 	strategy_x = BoundaryFactory::createBoundary(boundary_x);
 	strategy_y = BoundaryFactory::createBoundary(boundary_y);
 	strategy_z = BoundaryFactory::createBoundary(boundary_z);
+
+    boundaryConditionIndicator[0] = boundary_x == "Periodic" ? 1 : 0;
+    boundaryConditionIndicator[1] = boundary_y == "Periodic" ? 1 : 0;
+    boundaryConditionIndicator[2] = boundary_z == "Periodic" ? 1 : 0;

```

Comment:
Would it be better to declare labels for all boundary condition types and assign them here instead?

Reply:
Agree, we can make a bigger PR to fix hardcoded strings

Reply:
added new issue to address it https://github.com/CompuCell3D/CompuCell3D/issues/358